### PR TITLE
Tdb/the rest of your life edits

### PR DESCRIPTION
**Pull request by trevordblack**
As a means of procrastinating on other edits that I'm working on, I decided to read back through TheRestOfYourLife.

This is an omni-bus PR for all of the changes I found in chapters 1-6.
Ideally, lots of these would be within unique Issues and PRs, but I pushed for velocity over cleanliness.

Don't be kind.
I'll loop back around and create more PRs for other problems I perceive in the later chapters.

### CODE REVIEW

File: books/RayTracingTheRestOfYourLife.html [hunk 1]
```diff
@@ -968,7 +968,7 @@
             auto x = cos(2*pi*r1)*2*sqrt(r2*(1-r2));
             auto y = sin(2*pi*r1)*2*sqrt(r2*(1-r2));
             auto z = 1 - 2*r2;
-            std::cout << x << " " << y << " " << z << '\n';
+            printf("%f %f %f\n", x, y, z);
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html]: Stahp.

File: books/RayTracingTheRestOfYourLife.html [hunk 2]
```diff
@@ -1018,8 +1018,8 @@
             auto z = 1 - r2;
             sum += z*z*z / (1.0/(2.0*pi));
         }
-        std::cout << "PI/2 = " << pi/2 << '\n';
-        std::cout << "Estimate = " << sum/N << '\n';
+        printf("Pi/2     = %1.8f\n", pi/2);
+        printf("Estimate = %1.8f\n", sum/N);
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html]: Quit trying to make C happen.

File: books/RayTracingTheRestOfYourLife.html [hunk 3]
```diff
@@ -1065,10 +1065,10 @@
         auto sum = 0.0;
         for (int i = 0; i < N; i++) {
             vec3 v = random_cosine_direction();
-            sum += v.z()*v.z()*v.z() / (v.z()/(pi));
+            sum += v.z()*v.z()*v.z() / (v.z()/pi);
         }
-        std::cout << "PI/2 = " << pi/2 << '\n';
-        std::cout << "Estimate = " << sum/N << '\n';
+        printf("Pi/2     = %1.8f\n", pi/2);
+        printf("Estimate = %1.8f\n", sum/N);
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html]: Leave as is.

File: books/RayTracingTheRestOfYourLife.html [hunk 4]
```diff
@@ -1110,25 +1112,27 @@
...
 
 <div class='together'>
-Some models will come with at least one cotangent vector. The hard case of making an ONB is when we
-just have one vector. Suppose we have any vector $\vec{\mathbf{a}}$ that is not zero length or
-parallel to $\vec{\mathbf{n}}$. We can get to vectors $\vec{\mathbf{s}}$ and $\vec{\mathbf{t}}$
-parallel to $\vec{\mathbf{n}}$ by using the property of the cross product that
-$\vec{\mathbf{c}} \times \vec{\mathbf{d}}$ is perpendicular to both $\vec{\mathbf{c}}$ and
-$\vec{\mathbf{d}}$:
+Some models will come with one or more cotangent vectors. If our model has only one cotangent
+vector then the process of making an ONB is a nontrivial one. Suppose we have any vector
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html]: vector, then

File: books/RayTracingTheRestOfYourLife.html [hunk 5]
```diff
@@ -1110,25 +1112,27 @@
...
+$\vec{\mathbf{a}}$ that is of nonzero length and that is nonparallel to $\vec{\mathbf{n}}$. We can
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html]: of nonzero length and ~that is~ nonparallel to

File: books/RayTracingTheRestOfYourLife.html [hunk 6]
```diff
@@ -1149,9 +1149,9 @@
 $random(x,y,z)$ relative to the Z-axis we can get the vector relative to $\vec{n}$ as:
 
   $$
-  \text{Random vector} = (\vec{\mathbf{x}} \cdot \vec{\mathbf{s}})
-                       + (\vec{\mathbf{y}} \cdot \vec{\mathbf{t}})
-                       + (\vec{\mathbf{z}} \cdot \vec{\mathbf{n}})
+  \text{Random vector} = (x \cdot \vec{\mathbf{s}})
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html]: Eliminate `\cdot` and the parentheses, collapse to one line.

File: books/RayTracingTheRestOfYourLife.html [hunk 7]
```diff
@@ -1236,10 +1236,11 @@
 
 <div class='together'>
 It’s really easy to pick a random direction toward the light; just pick a random point on the light
-and send a ray in that direction. But we also need to know the PDF(direction). What is that? For a
-light of area $A$, if we sample uniformly on that light, the PDF on the surface of the light is
-$1/A$. But what is it on the area of the unit sphere that defines directions? Fortunately, there is
-a simple correspondence, as outlined in the diagram:
+and send a ray in that direction. But we also need to know the PDF, $p(direction)$. What is that?
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html]: direction, but we also

File: books/RayTracingTheRestOfYourLife.html [hunk 8]
```diff
@@ -1236,10 +1236,11 @@
...
+
+For a light of area $A$, if we sample uniformly on that light, the PDF on the surface of the light
+is $1/A$. But what is it on the area of the unit sphere that defines directions? Fortunately, there
```
Comment (hollasch) [inline books/RayTracingTheRestOfYourLife.html]: `$\frac{1}{A}$`

**Comment by trevordblack** [comment]
Well, you can't say that I wasn't consistent about printf (except where I wasn't)

File: src/TheRestOfYourLife/pdf.h [hunk 9]
```diff
@@ -47,6 +47,14 @@ vec3 random_on_unit_sphere() {
     return unit_vector(random_in_unit_sphere());
 }
 
+vec3 random_in_hemisphere(const vec3& normal) {
+    vec3 in_unit_sphere = random_in_unit_sphere();
+    if (dot(in_unit_sphere, normal) > 0.0) // In the same hemisphere as the normal
```
[the remark this answers is not in the document]
Comment (trevordblack) [reply src/TheRestOfYourLife/pdf.h]: Is this in relation to #270?

**Pull request merged**